src/Uno.UI/UI/Xaml/DependencyObjectStore.cs

```diff
@@ -1911,7 +1912,7 @@ DependencyPropertyDetails propertyDetails
 			{
 				if (value?.GetType() != propertyDetails.Property.Type)
 				{
-					value = Convert.ChangeType(value, propertyDetails.Property.Type);
+					value = Convert.ChangeType(value, propertyDetails.Property.Type, CultureInfo.InvariantCulture);
```

Comment:
This one may be problematic. If we convert a double from the user, invariant may not work. Let's keep `CurrentUICulture`, but we'll have to validate that further.